File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left=0
        right=len(numbers)-1
        res=[0,0]

        while left<right:
            if numbers[left]+numbers[right]==target:
                res[0]=left+1
                res[1]=right+1
                return res
            elif numbers[left]+numbers[right]<target:
                left+=1
            else:
                right-=1
        return res

        

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) 
    {
        List<Integer> res = new ArrayList<>();
        Deque<TreeNode> q = new ArrayDeque<>();
        if(root==null) return res;
        q.offer(root);
        while(!q.isEmpty())
        {
            int s = q.size();
            int t=0;
            while(t<s)
            {
                TreeNode temp = q.poll();
                if(t==s-1)
                {
                res.add(temp.val);
                }
                if(temp.left!=null) q.offer(temp.left);
                if(temp.right!=null) q.offer(temp.right);
                t++;
            }
        }
       return res; 
    }
}
